app/models/parameter.rb

```diff
@@ -10,6 +10,29 @@ class Parameter < ActiveRecord::Base
   validates :name, :presence => true, :no_whitespace => true
 
   scoped_search :on => :name, :complete_value => true
+  scoped_search :on => :type, :complete_value => true
+  scoped_search :on => :value, :complete_value => true
+
+  # children associations must be defined here, otherwise scoped search definitions won't find them
+  belongs_to :domain, :foreign_key => :reference_id, :inverse_of => :domain_parameters
```

Comment:
I think you misinterpreted - I meant to call `superclass.belongs_to` and `superclass.scoped_search` inside the subclasses. I'm not certain it's a good idea, but it might be preferable to keeping the child class associations in the parent class file.